src/internal/group_api/mod.rs

```diff
@@ -526,6 +530,82 @@ pub async fn group_create<CR: rand::CryptoRng + rand::RngCore>(
     resp.try_into()
 }
 
+#[derive(Debug, Clone)]
+pub struct GroupUpdatePrivateKeyResult {
+    needs_rotation: bool,
+}
+
+impl GroupUpdatePrivateKeyResult {
+    /// True if this group's private key requires additional rotation
+    pub fn needs_rotation(&self) -> bool {
+        self.needs_rotation
+    }
+}
+
+/// Rotate the group's private key. The public key for the group remains unchanged.
+pub async fn group_rotate_private_key<CR: rand::CryptoRng + rand::RngCore>(
+    recrypt: &Recrypt<Sha256, Ed25519, RandomBytes<CR>>,
+    group_id: &GroupId,
+    auth: &RequestAuth,
+    device_private_key: &PrivateKey,
+) -> Result<GroupUpdatePrivateKeyResult, IronOxideErr> {
+    let group_info = group_get_request(&auth, group_id).await?;
+    let admins = match &group_info.admin_ids {
+        Some(user_id_strings) => user_id_strings
+            .into_iter()
+            .map(|string_id| string_id.to_owned().try_into())
+            .collect::<Result<Vec<_>, _>>(),
+        None => Err(IronOxideErr::NotGroupAdmin(group_id.to_owned())),
+    }?;
+    let (_, admins_with_keys) = get_user_keys(auth, &admins).await?;
```

Comment:
I added a similar comment to Tonias PR, but we should probably check that we got all of the admins keys that we asked for and fail if we didn't

https://github.com/IronCoreLabs/ironweb/pull/65/files#r354552519

Reply:
Colt commented on Tonias and corrected me that the server will reject if the list of keys isn't the same, so I was wrong here. It doesn't hurt anything to check the list, but it's not necessary. So up to you to either leave it in or remove this. Sorry I led you astray.